src/_cffi_src/openssl/src/osrandom_engine.c

```diff
@@ -90,9 +94,54 @@ static struct {
     ino_t st_ino;
 } urandom_cache = { -1 };
 
+static int set_cloexec(int fd) {
+    int flags = fcntl(fd, F_GETFD);
+    if (flags == -1) {
+        return -1;
+    }
+    if (fcntl(fd, F_SETFD, flags | FD_CLOEXEC) == -1) {
+        return -1;
+    }
+    return 0;
+}
+
+static void close_fd(int fd) {
+    int n;
+    do {
+        n = close(fd);
+    } while (n < 0 && errno == EINTR);
+}
+
+#ifdef __linux__
+/* On Linux, we open("/dev/random") and use select() to wait until it's readable
+ * before we read from /dev/urandom, this ensures that we don't read from
+ * /dev/urandom before the kernel CSPRNG is initialized. This isn't necessary on
+ * other platforms because they don't have the same _bug_ as Linux does with
+ * /dev/urandom and early boot. */
+static int wait_on_devrandom(void) {
+    struct pollfd pfd = { 0 };
```

Comment:
`= {}` suffices to zero-initialize, and is IMO less confusing than `= { 0 }` since `struct pollfd` has three members.